src/main/java/Controller/GetChat.java

```diff
@@ -0,0 +1,71 @@
+package Controller;
+
+import Entity.Chat;
+import DBAccesser.User.QueryUser;
+import DBAccesser.Chat.QueryChat;
+import DBAccesser.UserChat.QueryUserChat;
+import Helper.SuccessResponseGenerator;
+import Exceptions.UserIDDoesNotExistException;
+import Exceptions.ChatIDDoesNotExistException;
+import Exceptions.UserChatIDDoesNotExistException;
+import Exceptions.UserIDMissingFromRequestURLPathException;
+import Exceptions.ChatIDMissingFromRequestURLParameterException;
+
+import java.util.Map;
+import java.util.LinkedHashMap;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.PathVariable;
+
+@RestController
+public class GetChat {
+    
+    @Autowired
+    QueryUser queryUserHelper;
+
+    @Autowired
+    QueryChat queryChatHelper;
+
+    @Autowired
+    QueryUserChat queryUserChatHelper;
+
+    @GetMapping("/users/chat")
+    public void getChatWithoutUserIDPathVariable() {
+
+        String path = "/users/chat";
+
+        throw new UserIDMissingFromRequestURLPathException(path);
+    }
+
+    @GetMapping("/users/{userID}/chat")
+    public Map<String, Object> getChat(@PathVariable("userID") String userIDString, @RequestParam(value = "chatID", required = false) String chatIDString) {
+
+        String path = "/users/" + userIDString + "/chat/?chatID="+chatIDString;
+        Map<String, Object> responseBody;
+
+        //check if URL parameter has been provided
+        if (chatIDString == null) {
+            throw new ChatIDMissingFromRequestURLParameterException(path);
+        }
+
+        //check if the passed userID is valid
+        if (queryUserHelper.checkIfUserIDExists(Long.parseLong(userIDString)) == false) {
+            throw new UserIDDoesNotExistException(path);
+        }
+        
+        //check if the passed chatID is valid
+        if (queryChatHelper.checkIfChatIDExists(Long.parseLong(chatIDString)) == false) {
+            throw new ChatIDDoesNotExistException(path);
+        } 
```

Comment:
`queryUser.checkIfUserIDExists(userID)` checks for the existence of the `User` in the `User` table
`queryUser.checkIfChatIDExists(chatID)` checks for the existence of the `Chat` in the `Chat` table
`queryUser.checkIfUserIDChatIDExists(userID, chatID)` checks for the existence of the `UserChat` in the `UserChat` table i.e. whether that `User` is part of that `Chat`

No, the read below doesn't miss any scenario. I added three separate checks so that the error message could be returned appropriately - 
```
{
    "timestamp": "2020-06-08T15:28:09.202246",
    "status": 404,
    "error": "404 NOT_FOUND",
    "message": ["UserID Does Not Exist"| "ChatID Does Not Exist" | "User is not part of this Chat"],
    "path": "/users/344145348288988520/chats/2065547810180608314"
}
```
- making it more convenient for the client to detect the source of error.

Reply:
Ack